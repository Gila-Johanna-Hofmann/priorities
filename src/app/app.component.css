

.content {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.header-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.input-container {
    margin-bottom: 5rem;
}

.toshop-container {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.shopped-container {
    margin-top: 5rem;
}

.one-list {
    display: flex;
    flex-direction: column;
    align-items: center;
}