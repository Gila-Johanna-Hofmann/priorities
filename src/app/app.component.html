<style></style>
<div class="content">
  <div class="header-container">
    <h1>{{ title }}</h1>
    <h3>{{ description }}</h3>
  </div>
  <Button *ngIf="isButtonHidden" (click)="onShowAddButton(false)"
    >Add Item</Button
  >
  <div class="input-container" *ngIf="!isButtonHidden">
    <app-input-form
      [shoppingItem]="emptyItem"
      [categories]="categories"
      (submitItem)="onSubmit($event)"
    ></app-input-form>
  </div>
  <div class="toshop-container">
    <div class="one-list">
      <app-need-list
        [shoppingList]="needList"
        (purchased)="onPurchasedChange($event)"
      ></app-need-list>
    </div>
    <div class="one-list">
      <app-want-list
        [shoppingList]="wantList"
        (purchased)="onPurchasedChange($event)"
      ></app-want-list>
    </div>
  </div>
  <div class="shopped-container">
    <div class="one-list">
      <app-purchased-list
        [shoppingList]="purchasedList"
      ></app-purchased-list>
    </div>
  </div>
</div>
