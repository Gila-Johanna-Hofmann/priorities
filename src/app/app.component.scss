h1 {
    color: #f7dd60;
}

h3 {
    color: white;
}

.content {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.header-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.input-container {
    margin-bottom: 5rem;
}

.toshop-container {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.shopped-container {
    margin-top: 5rem;
}

.one-list {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.save-button {
    display: flex;
    margin-top: 1rem;
    width: 100%;
    justify-content: center;

    button {
        min-width: 15rem;
    }
}