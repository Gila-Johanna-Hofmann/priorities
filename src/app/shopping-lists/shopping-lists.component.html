<p>shopping-lists works!</p>
<div class="shoppinglist"></div>
