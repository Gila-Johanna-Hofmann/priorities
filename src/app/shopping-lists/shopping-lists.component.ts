import { Component, OnInit, Input, Output } from '@angular/core';

@Component({
  selector: 'app-shopping-lists',
  templateUrl: './shopping-lists.component.html',
  styleUrls: ['./shopping-lists.component.css']
})
export class ShoppingListsComponent implements OnInit {

  
  

  constructor() { 
    
  }

  ngOnInit(): void {
  }

}
