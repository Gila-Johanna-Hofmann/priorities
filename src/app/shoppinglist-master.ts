export let shoppinglistMaster: ShoppingItem[] = [
  {
    title: 'TEST_Backpack',
    url: 'https://www.pinqponq.com/de/collections/sale/products/blok-large-vivid-monochrome',
    price: 45.55,
    category: 'need',
    purchased: false,
  },
  {
    title: 'TEST_Shoe',
    url: 'https://www.alohas.io/products/habibi-black-leather?variant=32485317247054',
    price: 1045.55,
    category: 'want',
    purchased: false,
  },
  {
    title: 'TEST_Boot',
    url: 'https://www.drmartens.com/de/de/p/21045001',
    price: 90.55,
    category: 'need',
    purchased: false,
  },
  {
    title: 'TEST_Bag',
    url: 'https://www.esthe.co.uk/product/beige-crossbody-bag/',
    price: 19,
    category: 'want',
    purchased: true,
  },
  {
    title: 'TEST_Hoodie',
    url: 'https://www.esthe.co.uk/product/beige-crossbody-bag/',
    price: 19,
    category: 'need',
    purchased: true,
  },
];
 
export interface ShoppingItem {
  title: string;
  url: string;
  price: number;
  category: string;
  purchased: boolean;
}
