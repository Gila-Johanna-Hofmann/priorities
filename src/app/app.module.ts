import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { NeedListComponent } from './need-list/need-list.component';
import { InputFormComponent } from './input-form/input-form.component';
import { FormsModule } from '@angular/forms';


@NgModule({
  declarations: [
    AppComponent,
    NeedListComponent,
    InputFormComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
