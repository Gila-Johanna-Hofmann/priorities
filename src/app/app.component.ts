import { Component, OnInit } from '@angular/core';
import { ShoppingService } from './Services/shopping.service';
import { ShoppingItem, uuidv4 } from './shoppinglist-master';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  title = 'priorities';

  description = 'keep track of stuff you need or want to buy';

  categories: string[] = [];

  needList: ShoppingItem[] = [];

  wantList: ShoppingItem[] = [];

  purchasedList: ShoppingItem[] = [];

  shoppingItem = {} as ShoppingItem;

  isButtonHidden = true;

  constructor(private shoppingService: ShoppingService) {}

  ngOnInit(): void {
    this.onRestoreFromLocalStorage()
  }

  initLists() {
    this.needList = this.shoppingService.getList(false, 'need');
    this.wantList = this.shoppingService.getList(false, 'want');
    this.purchasedList = this.shoppingService.getList(true);
    this.categories = this.shoppingService.categories;
  }

  onPurchasedChange(shoppingItem: ShoppingItem) {
    shoppingItem.purchased = true;
    this.initLists();
  }

  onSubmit(item: ShoppingItem) {
    if (!item.guid) {
      item = {
        ...item,
        purchased: false,
        category: item.category.toLowerCase(),
        guid: uuidv4(),
      };
      this.shoppingService.addItem(item);
    } else {
      item = { ...item, category: item.category.toLowerCase() };
      this.shoppingService.editItem(item);
    }
    this.onReset();
  }

  onShowAddButton(showButton: boolean) {
    this.isButtonHidden = showButton;
  }

  onReset() {
    this.initLists();
    this.isButtonHidden = true;
    this.shoppingItem = {} as ShoppingItem;
  }

  onEdit($event: ShoppingItem) {
    this.shoppingItem = { ...$event };
    this.isButtonHidden = false;
  }

  onDelete($event: string) {
    this.shoppingService.deleteItem($event);
    this.onReset();
  }

  onSaveList() {
      this.shoppingService.saveListToStorage();
  }

  onRestoreFromLocalStorage() {
    const restoredList = localStorage.getItem('shoppingList');
    if (restoredList && restoredList.length) {
      this.shoppingService.restoreList(JSON.parse(restoredList));
    }
    this.initLists();
  }
}
