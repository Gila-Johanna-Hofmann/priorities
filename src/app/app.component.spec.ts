import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { CATEGORIES_EXAMPLE, NEED_LIST_EXAMPLE, PURCHASED_LIST_EXAMPLE, WANT_LIST_EXAMPLE } from '../testing/mocks/shopping-lists.mock';
import { AppComponent } from './app.component';
import { ShoppingService } from './Services/shopping.service';
import { ShoppingItem } from './shoppinglist-master';

const LOCAL_STORAGE_SHOPPING_LIST: ShoppingItem[] = [{
    'guid': '374e36fd-5212-47a9-99e3-e479bad569a2',
    'title': 'TEST_Backpack',
    'url': 'https://www.pinqponq.com/de/collections/sale/products/blok-large-vivid-monochrome',
    'price': 45.55,
    'category': 'need',
    'purchased': false
}, {
    'guid': '75bd2588-683d-4bca-b4f2-76a25512ce85',
    'title': 'TEST_Shoe',
    'url': 'https://www.alohas.io/products/habibi-black-leather?variant=32485317247054',
    'price': 1045.55,
    'category': 'want',
    'purchased': false
}, {
    'guid': '5f8670f4-41ae-4353-980c-83ad87927465',
    'title': 'TEST_Boot',
    'url': 'https://www.drmartens.com/de/de/p/21045001',
    'price': 90.55,
    'category': 'need',
    'purchased': false
}, {
    'guid': '92629170-d778-46cc-8f38-c6f64cf4713c',
    'title': 'TEST_Bag',
    'url': 'https://www.esthe.co.uk/product/beige-crossbody-bag/',
    'price': 19,
    'category': 'want',
    'purchased': true
}, {
    'guid': 'b3a6ef65-5fd0-49ef-bb89-de255a9352f0',
    'title': 'TEST_Hoodie',
    'url': 'https://www.esthe.co.uk/product/beige-crossbody-bag/',
    'price': 19,
    'category': 'need',
    'purchased': true
}];

describe('AppComponent', () => {
    let component: AppComponent;
    let fixture: ComponentFixture<AppComponent>;
    let shoppingService: ShoppingService;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [
                RouterTestingModule
            ],
            declarations: [
                AppComponent,
            ],
        }).compileComponents();
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(AppComponent);
        shoppingService = TestBed.inject(ShoppingService);
        component = fixture.componentInstance;
        fixture.detectChanges();
    });

    it('should match snapshot', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should init all shopping lists', () => {
        expect(component.needList).toEqual(NEED_LIST_EXAMPLE);
        expect(component.wantList).toEqual(WANT_LIST_EXAMPLE);
        expect(component.purchasedList).toEqual(PURCHASED_LIST_EXAMPLE);
        expect(component.categories).toEqual(CATEGORIES_EXAMPLE);
    });

    it('should restore local storage list', () => {
        localStorage.setItem('shoppingList', JSON.stringify(LOCAL_STORAGE_SHOPPING_LIST));
        component.onRestoreFromLocalStorage();
        expect(component.needList).toEqual(LOCAL_STORAGE_SHOPPING_LIST.filter((item: ShoppingItem) => item.category === 'need' && !item.purchased));
    });

    it('should change purchase status', () => {
        const shoppingItem = { ...NEED_LIST_EXAMPLE[0], purchased: true };
        component.onPurchasedChange(shoppingItem);

        expect(component.needList).toHaveLength(2);
    });

    it('should add new item to need list', () => {
        const shoppingItem = {title: 'Example', category: 'need'} as ShoppingItem
        component.onSubmit(shoppingItem);
        expect(component.needList).toHaveLength(3);
    });

    it('should update shopping item', () => {
        const shoppingItem = {...NEED_LIST_EXAMPLE[1], title: 'UPDATED'};
        component.onSubmit(shoppingItem);
        expect(component.needList).toContainEqual(shoppingItem);
    });

    it('should change add button visibility', () => {
        component.onShowAddButton(true);
        expect(component.isButtonHidden).toBeTruthy();
    });

    it('should call on edit function', () => {
        const shoppingItem = {...NEED_LIST_EXAMPLE[1], title: 'UPDATED'};
        component.onEdit(shoppingItem);
        expect(component.shoppingItem).toEqual(shoppingItem);
        expect(component.isButtonHidden).toBeFalsy();
    });

    it('should delete an shopping item', () => {
        component.onDelete(NEED_LIST_EXAMPLE[1].guid);
        expect(component.needList).toHaveLength(1);
    });

    it('should call on save list methode', () => {
        shoppingService.saveListToStorage = jest.fn();
        component.onSaveList();
        expect(shoppingService.saveListToStorage).toHaveBeenCalled();
    });
});
