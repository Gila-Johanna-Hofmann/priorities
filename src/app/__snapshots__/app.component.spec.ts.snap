// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<app-root
  __ngContext__={[Function LRootView]}
  categories={[Function Array]}
  description={[Function String]}
  isButtonHidden={[Function Boolean]}
  needList={[Function Array]}
  purchasedList={[Function Array]}
  shoppingItem={[Function Object]}
  shoppingService={[Function ShoppingService]}
  title={[Function String]}
  wantList={[Function Array]}
>
  <div
    class="content"
  >
    <div
      class="header-container"
    >
      <h1>
        priorities
      </h1>
      <h3>
        keep track of stuff you need or want to buy
      </h3>
    </div>
    <button>
      Add Item 
    </button>
    <div
      class="toshop-container"
    >
      <div
        class="one-list"
      >
        <app-shopping-item-list />
      </div>
      <div
        class="one-list"
      >
        <app-shopping-item-list />
      </div>
    </div>
    <div
      class="shopped-container"
    >
      <div
        class="one-list"
      >
        <app-shopping-item-list />
      </div>
    </div>
  </div><div
    class="save-button"
  >
    <button>
       Save 
    </button>
  </div>
</app-root>
`;
